packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/NetCDFService.java

```diff
@@ -0,0 +1,133 @@
+package software.uncharted.terarium.hmiserver.service.data;
+
+import co.elastic.clients.elasticsearch.core.SearchRequest;
+import com.google.common.collect.ImmutableList;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import software.uncharted.terarium.hmiserver.configuration.Config;
+import software.uncharted.terarium.hmiserver.configuration.ElasticsearchConfiguration;
+import software.uncharted.terarium.hmiserver.models.dataservice.NetCDF;
+import software.uncharted.terarium.hmiserver.models.dataservice.PresignedURL;
+import software.uncharted.terarium.hmiserver.service.elasticsearch.ElasticsearchService;
+import software.uncharted.terarium.hmiserver.service.s3.S3ClientService;
+import ucar.ma2.Array;
+import ucar.nc2.Attribute;
+import ucar.nc2.NetcdfFile;
+import ucar.nc2.NetcdfFiles;
+import ucar.unidata.io.InMemoryRandomAccessFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class NetCDFService {
```

Comment:
We've moved to centralizing all of the service layer code between the controllers and ES, so this class should now just extend `S3BackedAssetService`. As an example, see `DatasetService`.

Reply:
Always a joy when a major refactor sweeps through